src/components/common/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,75 @@
+import useEmblaCarousel, { EmblaOptionsType } from 'embla-carousel-react';
+import { FC, ReactNode, useCallback, useEffect, useState } from 'react';
+
+type CarouselProps = {
+  children: ReactNode[];
+  dotButton?: boolean;
+  options?: EmblaOptionsType;
+};
+
+export const Carousel: FC<CarouselProps> = ({
+  children,
+  options = {},
+  dotButton = false,
+}) => {
+  const [emblaRef, embla] = useEmblaCarousel(options);
```

Comment:
[imo]
参考にしたリンク先でも同じような書き方なのですが、この変数名だと何のパラメータなの？って思うかなと感じたので以下のような変数名の方が良い気がしました。

```suggestion
  const [viewportRef, carousel] = useEmblaCarousel(options);
```

```suggestion
  const [viewportRef, carouselControl] = useEmblaCarousel(options);
```